pkg/hpa/service.go

```diff
@@ -765,3 +766,54 @@ func (c *Service) excludeExternalMetric(ctx context.Context, hpa *v2.HorizontalP
 
 	return newHPA
 }
+
+func (c *Service) CheckHpaMetricStatus(ctx context.Context, currenthpa *v2.HorizontalPodAutoscaler) bool {
```

Comment:
Actually, can we do like this instead? It would eliminate the logic from the controller layer.

```go
func (c *Service) UpdateTortoisePhaseIfHPAIsUnhealthy(ctx context.Context, currenthpa *v2.HorizontalPodAutoscaler, tortoise *v1beta1.Tortoise) error {
...

			if condition.Type == "ScalingActive" && condition.Status == "False" && condition.Reason == "FailedGetResourceMetric" {
				//switch to Emergency mode since no metrics
				logger.Info("HPA failed to get resource metrics, switch to emergency mode")
				tortoise.Status.TortoisePhase = v1beta3.TortoisePhaseEmergency
				return nil
			}

}
```

Then, probably it'd make more sense to move inside the tortoise service, instead of hpa service.

Reply:
Wont this add extra responsibilities to tortoise service if the service has to manage hpa conditions as well? Would it be better to add a `UpdateTortoisePhaseIfHPAIsUnhealthy` function in tortoise service but still perform the check in hpa service? so we only pass in the scalingActive bool into tortoise service instead of the entire hpa. Im not sure which will be cleaner

Reply:
> Wont this add extra responsibilities to tortoise service if the service has to manage hpa conditions as well

Hmm, that's understandable. So,

> Would it be better to add a UpdateTortoisePhaseIfHPAIsUnhealthy function in tortoise service but still perform the check in hpa service? so we only pass in the scalingActive bool into tortoise service instead of the entire hpa.

this direction looks good.